Compile error redefining UCS_BIT and some types

At commit 551d453caaac62b66769cbaba6b797b5d3b49338, master branch

Problem seen on CentOS 6.6 (gcc 4.4.7) and Fedora 25 (gcc 6.3.1)

(RHEL7.1 was ok, gcc 4.8.5)


```
make[2]: Entering directory `/gpfs/home/arcurtis/src/ucx/master/build/src/tools/perf'
libtool: compile:  gcc -DHAVE_CONFIG_H -I../../.. -O3 -g -Wall -Werror -I/gpfs/home/arcurtis/opt/ucx/include -mavx -I/opt/knem-1.1.1.90mlnx/include "-DCPU_FLAGS=|avx" -I/opt/knem-1.1.1.90mlnx/include -I/gpfs/home/arcurtis/src/ucx/master/build/../source/src -I/gpfs/home/arcurtis/src/ucx/master/build/src -fopenmp ../../../../source/src/tools/perf/perftest.c -c  -fPIC -DPIC -o .libs/perftest.o
In file included from ../../../../source/src/tools/perf/libperf.h:19,
                 from ../../../../source/src/tools/perf/perftest.c:15:
/gpfs/home/arcurtis/opt/ucx/include/ucs/sys/math.h:79:1: error: "UCS_BIT" redefined
In file included from /gpfs/home/arcurtis/opt/ucx/include/ucs/type/status.h:12,
                 from /gpfs/home/arcurtis/opt/ucx/include/uct/api/uct_def.h:10,
                 from /gpfs/home/arcurtis/opt/ucx/include/uct/api/uct.h:12,
                 from ../../../../source/src/tools/perf/libperf.h:17,
                 from ../../../../source/src/tools/perf/perftest.c:15:
/gpfs/home/arcurtis/opt/ucx/include/ucs/sys/compiler_def.h:32:1: error: this is the location of the previous definition
In file included from ../../../../source/src/tools/perf/libperf_int.h:16,
                 from ../../../../source/src/tools/perf/perftest.c:16:
/gpfs/home/arcurtis/opt/ucx/include/ucs/time/time.h:20: error: conflicting types for ‘ucs_time_t’
/gpfs/home/arcurtis/opt/ucx/include/ucs/time/time_def.h:18: note: previous declaration of ‘ucs_time_t’ was here
In file included from /gpfs/home/arcurtis/opt/ucx/include/ucs/async/signal.h:11,
                 from /gpfs/home/arcurtis/opt/ucx/include/ucs/async/async.h:11,
                 from ../../../../source/src/tools/perf/libperf_int.h:17,
                 from ../../../../source/src/tools/perf/perftest.c:16:
/gpfs/home/arcurtis/opt/ucx/include/ucs/datastruct/list.h:16: error: redefinition of ‘struct ucs_list_link’
/gpfs/home/arcurtis/opt/ucx/include/ucs/datastruct/list.h:19: error: conflicting types for ‘ucs_list_link_t’
/gpfs/home/arcurtis/opt/ucx/include/ucs/datastruct/list_types.h:17: note: previous declaration of ‘ucs_list_link_t’ was here
In file included from ../../../../source/src/tools/perf/libperf_int.h:17,
                 from ../../../../source/src/tools/perf/perftest.c:16:
/gpfs/home/arcurtis/opt/ucx/include/ucs/async/async.h:43: error: redefinition of typedef ‘ucs_async_event_cb_t’
/gpfs/home/arcurtis/opt/ucx/include/ucs/async/async_fwd.h:25: note: previous declaration of ‘ucs_async_event_cb_t’ was here
/gpfs/home/arcurtis/opt/ucx/include/ucs/async/async.h:81: error: conflicting types for ‘ucs_async_add_timer’
/gpfs/home/arcurtis/opt/ucx/include/ucs/async/async_fwd.h:64: note: previous declaration of ‘ucs_async_add_timer’ was here
make[2]: *** [perftest.lo] Error 1
make[2]: Leaving directory `/gpfs/home/arcurtis/src/ucx/master/build/src/tools/perf'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/gpfs/home/arcurtis/src/ucx/master/build'
make: *** [all] Error 2
```

--- a/src/tools/perf/Makefile.am
+++ b/src/tools/perf/Makefile.am
@@ -69,8 +69,8 @@ ucx_perftest_SOURCE_FILES = \
 perftest.lo: $(ucx_perftest_SOURCE_FILES)
 	$(AM_V_CC)
 	@$(LIBTOOL) $(AM_V_lt) --mode=compile --tag=CC $(ucx_perftest_CC) \
-		$(DEFS) -I$(top_builddir) $(CFLAGS) $(CPPFLAGS) $(AM_CPPFLAGS) $(ucx_perftest_CPPFLAGS) \
-		$(ucx_perftest_CFLAGS) $(ucx_perftest_SOURCE_FILES) -c -o $@
+		$(DEFS) -I. -I$(top_builddir) $(AM_CPPFLAGS) $(CPPFLAGS) $(ucx_perftest_CPPFLAGS) \
+		$(AM_CFLAGS) $(CFLAGS) $(ucx_perftest_CFLAGS) $(ucx_perftest_SOURCE_FILES) -c -o $@
 
 ucx_perftest$(EXEEXT): perftest.lo $(ucx_perftest_LDADD)
 	$(AM_V_CCLD)

